org.eclipse.jgit/src/org/eclipse/jgit/internal/storage/dfs/DfsBlockCaffeineCache.java

```diff
@@ -0,0 +1,165 @@
+package org.eclipse.jgit.internal.storage.dfs;
+
+import com.github.benmanes.caffeine.cache.*;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+public final class DfsBlockCaffeineCache extends DfsBlockCache {
+
+    /** Pack files smaller than this size can be copied through the cache. */
+    private final long maxStreamThroughCache;
+
+    /**
+     * Suggested block size to read from pack files in.
+     * <p>
+     * If a pack file does not have a native block size, this size will be used.
+     * <p>
+     * If a pack file has a native size, a whole multiple of the native size
+     * will be used until it matches this size.
+     * <p>
+     * The value for blockSize must be a power of 2.
+     */
+    private final int blockSize;
```

Comment:
Added a check in the config. Will switch to Immutable + checkArgument when we move the code >_<